nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -632,6 +628,35 @@ public void nestedTernary() {
         .doTest();
   }
 
+  @Test
+  public void genericsChecksForParameterPassing() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "static class A<T extends @Nullable Object> { }",
+            "  static A<String> sampleMethod(A<A<String>> a1, A<String> a2) {",
+            "     return a2;",
+            "  }",
+            "  static A<@Nullable String> sampleMethodWithVarArgs(A<A<@Nullable String>> a1, A<@Nullable String> a2, @Nullable Object o1, @Nullable Object o2) {",
```

Comment:
This is not using varargs.  You can read about varargs here:

https://www.baeldung.com/java-varargs